core/src/lib/jcr.cc

```diff
@@ -521,23 +510,24 @@ std::shared_ptr<JobControlRecord> GetJcrById(uint32_t JobId)
       [JobId](const JobControlRecord* jcr) { return jcr->JobId == JobId; });
 }
 
-std::shared_ptr<JobControlRecord> GetJcrByFullName(std::string name)
+std::shared_ptr<JobControlRecord> GetJcrByFullName(std::string_view name)
 {
-  return GetJcr([&name](const JobControlRecord* jcr) {
-    return std::string{jcr->Job} == name;
+  return GetJcr([name](const JobControlRecord* jcr) {
+    return std::string_view{jcr->Job} == name;
   });
 }
 
-std::shared_ptr<JobControlRecord> GetJcrByPartialName(std::string name)
+std::shared_ptr<JobControlRecord> GetJcrByPartialName(std::string_view name)
 {
-  return GetJcr([&name](const JobControlRecord* jcr) {
-    return std::string{jcr->Job}.find(name) == 0;
+  return GetJcr([name](const JobControlRecord* jcr) {
+    return std::string_view{jcr->Job}.find(name) == 0;
   });
 }
 
-std::shared_ptr<JobControlRecord> GetJcrBySession(const VolumeSessionInfo& vsi)
+std::shared_ptr<JobControlRecord> GetJcrBySession(VolumeSessionInfo vsi)
 {
-  return GetJcr([&vsi](const JobControlRecord* jcr) {
+  static_assert(sizeof(vsi) == 8);
```

Comment:
why do we need to check that?

Reply:
Just making sure that the struct does not get too big for pass-by-value.